src/databricks_ai_bridge/utils/vector_search.py

```diff
@@ -60,10 +61,38 @@ def is_databricks_managed_embeddings(self) -> bool:
         return self.is_delta_sync_index() and self.embedding_source_column.get("name") is not None
 
 
+@dataclass
+class RetrieverSchema:
+    text_column: str = None
+    doc_uri: Optional[str] = None
+    primary_key: Optional[str] = None
+    other_columns: Optional[List[str]] = None
+
+
+def get_metadata(columns: List[str], result: List[Any], retriever_schema, ignore_cols):
+    metadata = {}
+    for col, value in zip(columns[:-1], result[:-1]):
```

Comment:
Can we add a quick comment here explaining the precedence and hierarchy so its easier to understand in the future.